### ISSUE TITLE
About BitOps Calculation on ResNet18 W4A4

### ISSUE THREAD

**Issue by sunnyxiaohu** [1hunters/LIMPQ#3]
Hi, thanks for you wonderfull work.

In paper [LIMPQ](https://arxiv.org/pdf/2203.08368.pdf), the quantized model W4A4 ResNet18 is 33.05G
![image](https://github.com/1hunters/LIMPQ/assets/16660209/1ef34b3d-9537-4804-9005-634a4c6126b7)
In paper https://arxiv.org/pdf/2401.01543.pdf, the quantized mode W4A4 ResNet18 is 35.0G
![image](https://github.com/1hunters/LIMPQ/assets/16660209/a1c471e8-8f5a-466d-8a08-cf2acac2bea0).

I'm curious what causes the difference in calculations here?

ps. I also run the `search.py` in the repo, it possses around 35.0G BitOps.

**Comment by 1hunters** [comment]
Hi, 

Sorry for the late response due to the Lunar New Year holiday and happy new year to you!

Thank you for your interest in our work. This is due to BitOPs calculation of the first layer (the stem layer) with different kernel sizes and output channels. In our initial experiments, we use a variant of ResNet18 (a bit like the ResNet18d [here](https://github.com/huggingface/pytorch-image-models/blob/3df4ffe914f7969344fc728c0b8a9b8289b3baa9/timm/models/resnet.py#L1273)) with 70.5% Top-1 accuracy, where the kernel size and output channels of stem layer are smaller than the vanilla ResNet18 but the depth of the stem layer is increased. This causes the BitOPs of the stem layer to be slightly lower than the vanilla model. We align the baseline's BitOPs (e.g., PACT) to be the same as ours for both fair comparison and search convenience of quantization policies.

In our following papers, we re-run LIMPQ on Timm's pre-trained vanilla ResNet18 (i.e., 70.5% FP top1 accuracy) and got the same top-1 accuracy (because we do not search and quantize the first and last layers to mixed-precision). 

Let me know if you have further questions.

Best, 
Chen

**Comment by ganjuzhizihuai** [comment]
> Reference

Hi,
I had a problem training the model, can you share your previous indicators of ResNet18? Thank you very much.

**Comment by 1hunters** [comment]
I didn't save these indicators locally as they were trained on my internship's servers. I think maybe you could describe your training issue or consider using our provided ResNet50 indicators.
